From: Erik de Castro Lopo <erikd@mega-nerd.com>
Date: Mon, 13 Feb 2012 08:58:08 +0000 (+1100)
Subject: Fix -Werror=format-security errors.
X-Git-Url: http://git.xiph.org/?p=flac.git;a=commitdiff_plain;h=a4ffcc0239cbf7384146f0d0e6770e1dbb1b58c4

Fix -Werror=format-security errors.

Fixes "format not a string literal and no format arguments
 [-Werror=format-security]" errors.

Patch from Fabian Greffrath <fabian+debian@greffrath.com> via Debian.
Closes Debian bug #643377.
---

--- flac-1.2.1/src/test_grabbag/cuesheet/main.c~	2007-02-02 06:26:04.000000000 +0000
+++ flac-1.2.1/src/test_grabbag/cuesheet/main.c	2012-11-12 22:19:10.859497990 +0000
@@ -115,12 +115,12 @@
 	const char *usage = "usage: test_cuesheet cuesheet_file lead_out_offset [ cdda ]\n";
 
 	if(argc > 1 && 0 == strcmp(argv[1], "-h")) {
-		printf(usage);
+		printf("%s", usage);
 		return 0;
 	}
 
 	if(argc < 3 || argc > 4) {
-		fprintf(stderr, usage);
+		fprintf(stderr, "%s", usage);
 		return 255;
 	}
 
@@ -129,7 +129,7 @@
 		if(0 == strcmp(argv[3], "cdda"))
 			is_cdda = true;
 		else {
-			fprintf(stderr, usage);
+			fprintf(stderr, "%s", usage);
 			return 255;
 		}
 	}
diff --git a/src/test_grabbag/picture/main.c b/src/test_grabbag/picture/main.c
index b59afd1..a1880ae 100644
--- a/src/test_grabbag/picture/main.c
+++ b/src/test_grabbag/picture/main.c
@@ -211,12 +211,12 @@ int main(int argc, char *argv[])
 	const char *usage = "usage: test_pictures path_prefix\n";
 
 	if(argc > 1 && 0 == strcmp(argv[1], "-h")) {
-		printf(usage);
+		printf("%s", usage);
 		return 0;
 	}
 
 	if(argc != 2) {
-		fprintf(stderr, usage);
+		fprintf(stderr, "%s", usage);
 		return 255;
 	}
 
diff --git a/src/test_seeking/main.c b/src/test_seeking/main.c
index d1f11d6..589e859 100644
--- a/src/test_seeking/main.c
+++ b/src/test_seeking/main.c
@@ -418,7 +418,7 @@ int main(int argc, char *argv[])
 	static const char * const usage = "usage: test_seeking file.flac [#seeks] [#samples-in-file.flac] [file.raw]\n";
 
 	if (argc < 2 || argc > 5) {
-		fprintf(stderr, usage);
+		fprintf(stderr, "%s", usage);
 		return 1;
 	}
 
